### PR TITLE
Implemented Command usage counter

### PR DESCRIPTION
**Pull request by qjie7**
Screenshot 2019-10-19 at 1 13 13 PM

I have created a new csv file to store the amount of time of particular command have been used. This feature will be used to fulfil the #135 requirements , where i most likely will be providing the top 3 commands with simple shortcut keys to have the command to be carried out and these top 3 commands will likely to be auto adjusted due to the number of time of commands were used. 

Basically , this command counter currently will keep on adding/tracking the commands entered by user.  The only way to make it restart from 0 , is to delete the csv file contents manually.

### CODE REVIEW

File: src/main/java/duke/command/HelpCommand.java [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
+//    /**
+//     * .
+//     *
+//     * @return .
+//     */
+//    @Override
+//    public boolean isExit() {
+//        return false;
+//    }
+//}
```
Comment (HUANGXUANKUN) [inline src/main/java/duke/command/HelpCommand.java]: This is an non-related changes belong to other feature. It shouldnt be modified in this PR for review readability

File: src/main/java/duke/core/CommandManager.java [hunk 2]
```diff
@@ -109,7 +124,13 @@ public static Command manageCommand(String userInput) throws DukeException {
                 throw new DukeException("update command fails. " + e.getMessage());
             }
         case "bye":
-            return new ExitCommand();
+            ExitCommand exitCommand = new ExitCommand();
+            for (Map.Entry<String, Integer> entry : cmdFreqTable.entrySet()) {
+                System.out.println("Key = " + entry.getKey());
+                System.out.println("Value = " + entry.getValue());
+                System.out.println("====================================================");
+            }
```
Comment (HUANGXUANKUN) [inline src/main/java/duke/core/CommandManager.java]: This part is not belonged to commandManaging level. All these should be added in ExitCommand.execute.

Comment (HUANGXUANKUN) [reply src/main/java/duke/core/CommandManager.java]: In Duke, command.isExit is executed after command.execute.
```
        while (!isExit) {
            try {
                String fullCommand = ui.readCommand();
                ui.showLine();
                Command c = CommandManager.manageCommand(fullCommand);
                c.execute(patientTaskList, taskManager, patientManager,
                        ui, patientTaskStorage, taskStorage, patientStorage);
                isExit = c.isExit();
```

File: src/main/java/duke/command/UpdateTaskCommand.java [hunk 3]
```diff
@@ -37,7 +40,17 @@ public UpdateTaskCommand(String command) {
...
+                        CommandManager commandManager) throws DukeException {
+        this.hasBeenAddedBefore = true;
+        String commandName = this.getClass().getSimpleName();
+        if (!hasBeenAddedBefore) {
+            commandManager.getCmdFreqTable().put(commandName, 1);
+        }
+        int count = commandManager.getCmdFreqTable().containsKey(commandName)
+                    ? commandManager.getCmdFreqTable().get(commandName) : 0;
+        commandManager.getCmdFreqTable().put(commandName, count + 1);
+
```
Comment (HUANGXUANKUN) [inline src/main/java/duke/command/UpdateTaskCommand.java]: All these are not belong to command.execute. 
1.  The condition check for <FreqCommandHandler.hasBeenAddedBefore(getClass().getName())>
2.  The increment of count

There should be a specific class to handle. Do an abstraction.

File: src/main/java/duke/command/HelpCommand.java (hunk 1 above)
Comment (qjie7) [reply src/main/java/duke/command/HelpCommand.java]: Since the help command is not in use , so i commented them out first

File: src/main/java/duke/core/CommandManager.java (hunk 2 above)
Comment (qjie7) [reply src/main/java/duke/core/CommandManager.java]: This code is just for me to print out for checking ... basically now exitcommand will not be keep track by the command frequency since i think it is pointless to keep track of exit command.

File: src/main/java/duke/command/UpdateTaskCommand.java (hunk 3 above)
Comment (qjie7) [reply src/main/java/duke/command/UpdateTaskCommand.java]: I have shifted all of these to a function in our abstract command class.

File: src/main/java/duke/command/HelpCommand.java (hunk 1 above)
Comment (HUANGXUANKUN) [reply src/main/java/duke/command/HelpCommand.java]: It does not make sense as it is not belong to this PR. 
This PR should only involve file changes that belong to the FreqCounter.
A PR should only serve one purpose.

File: src/main/java/duke/core/CommandManager.java (hunk 2 above)
Comment (HUANGXUANKUN) [reply src/main/java/duke/core/CommandManager.java]: Junit test is there to serve the purpose. It should not appear in CommandManager which only responsible to manage command.

File: src/main/java/duke/command/AddPatientCommand.java [hunk 4]
```diff
@@ -4,6 +4,8 @@
 import duke.core.Ui;
 import duke.patient.Patient;
 import duke.patient.PatientManager;
+import duke.statistic.Counter;
+import duke.storage.CounterStorage;
```
Comment (HUANGXUANKUN) [inline src/main/java/duke/command/AddPatientCommand.java]: Unused import. Discard in next PR.
